CollAction/Startup.cs

```diff
@@ -120,9 +121,30 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerF
 
             if (env.IsProduction())
             {
-                app.UseRewriter(new RewriteOptions().AddRewriteHttpsProxyRule());
+                // Ensure our middleware handles proxied https, see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Proto
+                app.UseForwardedHeaders(new ForwardedHeadersOptions()
+                {
+                    ForwardedHeaders = ForwardedHeaders.XForwardedProto | ForwardedHeaders.XForwardedFor | ForwardedHeaders.XForwardedHost
+                });
+
+                app.UseSecurityHeaders(new HeaderPolicyCollection() // See https://www.owasp.org/index.php/OWASP_Secure_Headers_Project
+                    .AddStrictTransportSecurityMaxAgeIncludeSubDomains() // Add a HSTS header, making sure browsers connect to collaction + subdomains with https from now on
+                    .AddXssProtectionEnabled() // Enable browser xss protection
+                    .AddContentTypeOptionsNoSniff() // Ensure the browser doesn't guess/sniff content-types
+                    .AddReferrerPolicyStrictOriginWhenCrossOrigin() // Send a full URL when performing a same-origin request, only send the origin of the document to a-priori as-much-secure destination (HTTPS->HTTPS), and send no header to a less secure destination (HTTPS->HTTP) 
+                    .AddContentSecurityPolicy(cspBuilder =>
```

Comment:
You most likely need some exceptions here for Google Analytics, YouTube, Facebook plugins etc.